Fluid/Values/ArrayValue.cs

```diff
@@ -145,15 +145,22 @@ public override bool Equals(object obj)
             // The is operator will return false if null
             if (obj is ArrayValue otherValue)
             {
-                return Values.Equals(otherValue.Values);
+                return Equals(otherValue);
             }
 
             return false;
         }
 
         public override int GetHashCode()
         {
-            return Values.GetHashCode();
+            var hc = new HashCode();
+
+            foreach (var v in Values)
```

Comment:
because `Values` seems to unfortunately be `IReadOnlyList` a for loop would be more efficient here